File: Бодлогууд/untitled/src/main/java/org/example/Employee.java
package org.example;

import java.text.DateFormat;
import java.util.Date;
import java.util.Vector;

public class Employee extends Person{

    private String CompanyID;
    private String Title;
    private Date StartDate;

    Employee(){
         CompanyID  = "Unnamed";
         Title  = "Unnamed";
         StartDate = new Date();
    }
    Employee(String CompanyID, String Title, Date StartDate, String name, String ssnum, int age, JobDescription job, Division division){
        this.CompanyID = CompanyID;
        this.Title = Title;
        this.StartDate = StartDate;
        this.name = name;
        this.ssnum = ssnum;
        this.age = age;
        this.job.add(job);
        this.division.setDivisionName(division.getDivisionName());

    }

    Vector<Spouse> spouse = new Vector<Spouse>();
    Vector<Child> child = new Vector<Child>();
    Vector<JobDescription> job = new Vector<JobDescription>();
    Division division = new Division();

    void addJob(JobDescription j){
        this.job.add(j);
    }
    void addChild(Child son){
        this.child.add(son);
    }

    void setSpouse( Spouse love){
        if(spouse.size() == 1)
            System.out.println("Hostoi hvn bna shvvdee ");
        else
            spouse.add(love);
    }

    void setDivision(Division a){
        division.setDivisionName(a.getDivisionName());
    }

    // get
    String getCompanyID(){
        return  CompanyID;
    }
    String getTitle(){
        return  Title;
    }
    Date getStartDate(){
        return StartDate;
    }
    // set
    void setCompanyID(String CompanyID){
        this.CompanyID = CompanyID;
    }
    void setTitle(String Title){
        this.Title = Title;
    }
    void setStartDate(Date StartDate) {
    this.StartDate = StartDate;
    }

    void print(){
        System.out.println(getName());
        System.out.println(getSsnum());
        System.out.println(getAge());
        System.out.println(getCompanyID());
        System.out.println(getTitle());
        System.out.println(getStartDate());
        //System.out.println(spouse.elementAt(0));
        System.out.println(division.getDivisionName());
    }


}

File: Бодлогууд/Лаб9/Employee.cpp
#include "Employee.h"

 Employee() : Person(){
            companyID = new char[strlen("unknown") + 1];
            strcpy(companyID, "unknown");
            title = new char[strlen("unknown") + 1];
            strcpy(title, "unknown");
            startDate = new char[strlen("unknown") + 1];
            strcpy(startDate, "unknown");
            cout << "*Ajilchin nemegdsen*" << endl;
        }
        Employee(char *id, char *garchig, char*date, char *ner, char* num, int nas, JobDescription desc, Division d) : Person(ner, num, nas){
            companyID = new char[strlen(id) + 1];
            strcpy(companyID, id);
            title = new char[strlen(garchig) + 1];
            strcpy(title, garchig);
            startDate = new char[strlen(date) + 1];
            strcpy(startDate, date);
            job.push_back(desc);
            div.setDivisionName(d.getDivisionName());
            cout << "*Ajilchin nemegdsen*" << endl;
        }
        void addDesc(JobDescription desc){
            this->job.push_back(desc);
        }
        void setDiv(Division d){
            div.setDivisionName(d.getDivisionName());
        }
        void addChild(Child hvvhed){
            this->child.push_back(hvvhed);
        }
        int childCount(){
            return child.size();
        }
        void setSpouse(Spouse ehner){
            if(spouse.size() == 1)
                cout << "Ehnertei bn\n";
            else
                spouse.push_back(ehner);
        }
        char* getCompanyID(){
            return companyID;
        }
        char* getTitle(){
            return title;
        }
        char* getStartDate(){
            return startDate;
        }
        void setCompanyID(char *id){
            delete companyID;
            companyID = new char[strlen(id) + 1];
            strcpy(companyID, id);
        }
        void setTitle(char *garchig){
            delete title;
            title = new char[strlen(garchig) + 1];
            strcpy(title, garchig);
        }
        void setStartDate(char* start){
            delete startDate;
            startDate = new char[strlen(start) + 1];
            strcpy(startDate, start);
        }
        ~Employee(){
            delete companyID;
            delete title;
            delete startDate;
            cout << endl;
        }
        void print(){
            cout << "Ajiltan: " <<getName() << "-ii medeelel: \n" <<endl;
            cout << "SSNum: " << getSSNum() << endl;
            cout << "Age: " << getAge() << endl;
            cout << "CompanyID: " << companyID << endl;
            cout << "Title: " << title << endl;
            cout << "StartDate: " << startDate << endl;
            cout << "Spouse: " << spouse[1].getName() << endl;
            cout << "Child: " << childCount() << endl;
            cout << "Division" << div.getDivisionName() << endl;
        }